lib/Doctrine/ORM/AbstractQuery.php

```diff
@@ -417,11 +418,18 @@ public function processParameterValue($value)
             return $value;
         }
 
-        if (is_object($value) && $this->_em->getMetadataFactory()->hasMetadataFor(ClassUtils::getClass($value))) {
-            $value = $this->_em->getUnitOfWork()->getSingleIdentifierValue($value);
+        if (is_object($value)) {
+            try {
+                $value = $this->_em->getUnitOfWork()->getSingleIdentifierValue($value);
+
+                if ($value === null) {
+                    throw ORMInvalidArgumentException::invalidIdentifierBindingEntity();
+                }
+            } catch (MappingException $e) {
```

Comment:
I think the performance impact is not worth the bugfix. This would mean automatic file I/O for anyone having a file-based metadata driver configured.

That said, I have no other/better solutions right now :-\

This will probably have to stay in a limbo unless we fix `hasMetadataFor()` for good.